crates/nu-command/src/filesystem/rm.rs

```diff
@@ -212,18 +247,15 @@ fn rm(
             .expect("targets were empty"),
     );
 
-    let path = current_dir(engine_state, stack)?;
-
     let (mut target_exists, mut empty_span) = (false, call.head);
     let mut all_targets: HashMap<PathBuf, Span> = HashMap::new();
 
     for target in targets {
-        if path.to_string_lossy() == target.item
-            || path.as_os_str().to_string_lossy().starts_with(&format!(
-                "{}{}",
-                target.item,
-                std::path::MAIN_SEPARATOR
-            ))
+        if currentdir_path.to_string_lossy() == target.item
+            || currentdir_path
+                .as_os_str()
+                .to_string_lossy()
+                .starts_with(&format!("{}{}", target.item, std::path::MAIN_SEPARATOR))
```

Comment:
Instead of converting to string you could check the path directly https://doc.rust-lang.org/std/path/struct.Path.html#method.starts_with for more compact code and less allocations.

Reply:
emm, but it is not the logic by me though, but what this line is doing?

Reply:
Ah, I see, sorry. It checks whether a path starts with another path.